Lab5/README.md

```diff
@@ -2,19 +2,554 @@
 
 # Lab 5 - Azure Cognitive Services
 
-## Goal
+# Goal
 
-We will add a Azure Cognitive Services Vision to our ARM template. Modify the code of the web application to read one image (provided) from a Azure blob storage, and display the information returned by the Vision API. The image ( a dog or a cat) will be uploaded to the Blob storage using the Azure POrtal or the Azure Storage Explorer.
+We will add a Azure Cognitive Services Vision to our ARM template. Modify the code of the web application to read one image (provided) from a Azure blob storage, and display the information returned by the Vision API. The image ( a dog or a cat ) will be uploaded to the Blob storage using the Azure Portal or the Azure Storage Explorer.
 
-## Let's code!
+# Let's code!
 
-[Enter here a step by step tutorial]
+- [Add Azure components](#add-azure-components)
+  * [Modify the ARM template to add the Computer Vision service](#modify-the-arm-template-to-add-the-computer-vision-service)
+  * [Modify the ARM template to add the Blob Storage](#modify-the-arm-template-to-add-the-blob-storage)
+  * [Deploy the ARM templates to Azure ??](#deploy-the-arm-templates-to-azure---)
+  * [Getting the API keys for the API](#getting-the-api-keys-for-the-api)
+  * [Getting the blob storage connection string](#getting-the-blob-storage-connection-string)
+- [Upload the dog image into your storage account](#upload-the-dog-image-into-your-storage-account)
+- [Let's add computer vision to our web application](#let-s-add-computer-vision-to-our-web-application)
+  * [Adding the Computer Vision API library to the application](#adding-the-computer-vision-api-library-to-the-application)
+  * [Adding the Blob storage library to the application](#adding-the-blob-storage-library-to-the-application)
+  * [Using the API in code](#using-the-api-in-code)
+    + [Setting up the resources keys](#setting-up-the-resources-keys)
+    + [Add code to make use of the resource keys](#add-code-to-make-use-of-the-resource-keys)
+    + [Creating classes to consume the storage and computer vision api](#creating-classes-to-consume-the-storage-and-computer-vision-api)
+      - [BlobStorageManager](#blobstoragemanager)
+      - [ImageAnalyzer](#imageanalyzer)
+      - [Wiring all this into the Startup](#wiring-all-this-into-the-startup)
+  * [Creating the controller and view](#creating-the-controller-and-view)
+- [Reference](#reference)
 
+# Add Azure components
 
-## Reference
+Let's add the Cognitive Services to our deployment to be able to make use of it.
 
-## End
+## Modify the ARM template to add the Computer Vision service
+
+DO WE HAVE A "STARTING" ARM TEMPLATE?
```

Comment:
Yes we have an ARM template created in Lab 2.
It also add the Storage account @alainvezina is writing that part.

Reply:
Im planning to work on this weekend.  I have the basic. Should not be that long